src/rust/runtime/queue/mod.rs

```diff
@@ -44,7 +44,7 @@ impl IoQueueTable {
     /// When Demikernel is interposing system calls of the underlying OS
     /// this offset enables us to distinguish I/O queue descriptors from
     /// file descriptors.
-    const BASE_QD: usize = 1_000_000;
+    const BASE_QD: u32 = 1_000_000;
```

Comment:
At first I thought this number was arbitrary, but now I'm seeing that it is roughly 2^20, which apparently is the maximum number of file descriptors that modern Linux allows (requires explicit configuration as this isn't the default).  One can go beyond this yet on Linux, but it takes more than adjusting the well-documented way to increase the file descriptor limit.

If this value is chosen to match the Linux limitation, then it probably should be in the PAL.  It should probably also have a comment explaining why it is the value it is.

Reply:
This is a pending fix that I have not yet submitted. This PR is about changing the type of queue descriptors.

Reply:
Yes, I know.  I was asking these questions because the new queue descriptor type has fewer bits, but still has to be big enough to hold (essentially) both the file descriptor namespace and the actual queue descriptor namespace.  This BASE_QD value is what determines how much of the u32's namespace goes to each.

Reply:
Oh sure! Let me give you some extra information: we will decrease this value :). It should be less than `FD_SETSIZE` [[1](https://www.man7.org/linux/man-pages/man2/select.2.html)] to play nice with `epoll()`.